jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -101,40 +100,76 @@ def _mode_selected_changed(self, event={}):
             self.session.edit_subset_mode = self.mode_selected
     '''
 
-    def _get_region_definition(self, *args):
-        self.subset_definition = {}
+    def _unpack_nested_subset(self, subset_state):
+        '''
+        Navigate through the tree of subset states for composite
+        subsets made up of multiple regions.
+        '''
+        if isinstance(subset_state, CompositeSubsetState):
+            self._unpack_nested_subset(subset_state.state1)
+            self._unpack_nested_subset(subset_state.state2)
+        else:
+            if subset_state is not None:
+                self._get_subset_subregion_definition(subset_state)
+
+    def _get_subset_subregion_definition(self, subset_state, subset_op=""):
+        """
+        Get the type and parameters for a single region in the subset. Note that
+        the string type and operation (if in a composite subset) need to be stored
+        separately from the float parameters for display reasons.
+        """
+        subset_type = {}
+        if subset_op:
+            subset_type["Operation"] = subset_op
+        subset_definition = None
+
+        if isinstance(subset_state, RoiSubsetState):
+            subset_classname = subset_state.roi.__class__.__name__
+            if subset_classname == "CircularROI":
+                x, y = subset_state.roi.get_center()
+                subset_definition = {"X Center": x,
+                                     "Y Center": y,
+                                     "Radius": subset_state.roi.radius}
+
+            elif subset_classname == "RectangularROI":
+                subset_definition = {}
+                for att in ("Xmin", "Xmax", "Ymin", "Ymax"):
+                    subset_definition[att] = getattr(subset_state.roi, att.lower())
+
+            elif subset_classname == "EllipticalROI":
+                subset_definition = {"X Center": subset_state.roi.xc,
+                                     "Y Center": subset_state.roi.yc,
+                                     "X Radius": subset_state.roi.radius_x,
+                                     "Y Radius": subset_state.roi.radius_y}
+            subset_type["Subset type"] = subset_classname
+
+        elif isinstance(subset_state, RangeSubsetState):
+            subset_definition = {"Upper bound": subset_state.hi,
+                                 "Lower bound": subset_state.lo}
+            subset_type["Subset type"] = "Range"
+
+        if subset_definition is not None and subset_definition not in self.subset_definitions:
+            self.subset_definitions.append(subset_definition)
+            self.subset_types.append(subset_type)
```

Comment:
these appends might be the issue with traitlets not automatically updating.  Try with `self.subset_definitions = self.subset_definitions + [subset_definition]` (`+=` won't update either, iirc)

Reply:
Seems like that did the trick, thanks. I fell back on the old way we had of working around it because I couldn't remember at the time what the real problem with it was.

Reply:
Wait, I spoke too soon - I hadn't actually saved the change apparently 😆